src/connector/src/parser/unified/json.rs

```diff
@@ -261,7 +261,7 @@ impl JsonParseOptions {
                 DataType::Int16,
                 ValueType::I64 | ValueType::I128 | ValueType::U64 | ValueType::U128,
             ) => value.try_as_i16().map_err(|_| create_error())?.into(),
-
+            (DataType::Int16, ValueType::Bool) => (value.as_bool().unwrap() as i16).into(),
```

Comment:
may I know why add these parsing logic?

Reply:
Seems a side effect of misusing of `never` snapshot.mode. Deleted.